src/ext/java/org/opentripplanner/ext/geocoder/StopClusterMapper.java

```diff
@@ -55,28 +57,40 @@ Iterable<StopCluster> generateStopClusters(
     return Iterables.concat(deduplicatedStops, stations);
   }
 
-  StopCluster map(StopLocationsGroup g) {
+  LuceneStopCluster map(StopLocationsGroup g) {
     var modes = transitService.getModesOfStopLocationsGroup(g).stream().map(Enum::name).toList();
-    return new StopCluster(
+    var agencies = transitService
+      .getAgenciesForStopLocationsGroup(g)
+      .stream()
+      .map(s -> s.getId().toString())
+      .toList();
+    return new LuceneStopCluster(
       g.getId(),
       null,
       g.getName().toString(),
       toCoordinate(g.getCoordinate()),
-      modes
+      modes,
+      agencies
     );
   }
 
-  Optional<StopCluster> map(StopLocation sl) {
+  Optional<LuceneStopCluster> map(StopLocation sl) {
```

Comment:
> However the OTP geocode endpoint does: https://atl-rides-otp-v2.ibi-transit.com/otp/routers/default/geocode?query=600011

This query returns:
```json
[
  {
    "lat": 33.780654,
    "lng": -84.387333,
    "description": "West Peachtree Street at Peachtree Place (600011)",
    "id": "CobbLinc:974"
  },
  {
    "lat": 33.780654,
    "lng": -84.387333,
    "description": "West Peachtree Street at Peachtree Place (600011)",
    "id": "GwinnettCountyTransit:843"
  },
  {
    "lat": 33.780654,
    "lng": -84.387333,
    "description": "West Peachtree St at Peachtree Pl (600011)",
    "id": "Xpress:328"
  },
  {
    "lat": 33.537298,
    "lng": -84.575597,
    "description": "Renaissance Pkwy at 6000 (213739)",
    "id": "MARTA:99974123"
  }
]
```

From the response to the above `geocode` query, we would expect the Cobb and Gwinnett stops to be combined, for starters. The Xpress stop could be included eventually when a name heuristics is found.

> If there was a single, integrated feed in Atlanta that would contain these three agencies and a station/stop hierarchy, this would work automatically.

Because the code has access to stops from all loaded feeds, I think the better approach would be to pass an ` `originalStops` param to this `map` method, so that way we can collect the agencies using all the stops that were part of the original cluster. Thoughts @leonardehrenfried  @miles-grant-ibigroup ?

Reply:
We talked about this in an IBI-internal call and we agreed that we accept the shortcomings of this approach and will iterate on it.